Chocolate_Vending_Machine.java

```diff
@@ -0,0 +1,240 @@
+package Week_7;
+
+import java.util.*;
+
+public class Chocolate_Vending_Machine implements Cash_Inventory, Item_Inventory {
+	private HashMap<Coin, Integer> cash = new HashMap<>();
+	HashMap<Item, Integer> items = new HashMap<>();
+
+	public void initialize_CashInventory() {
+	
+		for (Coin c : Coin.values()) {
+			cash.put(c, 5);
+		}
+	}
+
+	public void initialize_CashInventory(int coins[]) {
+		int i = 0;
+		for (Coin c : Coin.values()) {
+			cash.put(c, coins[i]);
+			i++;
+
+		}
+	}
+
+	public void intialize_Item_Inventory() {
+
+		for (Item it : Item.values()) {
+			items.put(it, 5);
+
+		}
+	}
+
+	public void intialize_Item_Inventory(int item_values[]) {
+		int i = 0;
+		for (Item it : Item.values()) {
+			items.put(it, item_values[i]);
+			i++;
+
+		}
+	}
+
+	public void addCoin(Coin c, int no) {
+		int value = cash.get(c);
+		cash.put(c, value + no);
+		// totalCash+=(c.getDenomination()*no);
+	}
+
+	public void addItem(Item i, int no) {
+		int val = items.get(i);
+		items.put(i, val + no);
+	}
+
+	public void removeCoin(Coin c, int no) {
+		int value = cash.get(c);
+		if (value >= no) {
+			cash.put(c, value - no);
+		} else {
+			cash.put(c, 0);
+		}
+	}
+
+	public void removeItem(Item i, int no) throws OutOfStockException {
+		int val = items.get(i);
+		if (val >= no) {
+			items.put(i, val - no);
+		} else {
+			throw new OutOfStockException("Sorry!! We don't have enough chocolates to meet your demand");
+		}
+	}
+
+	Chocolate_Vending_Machine() {
+		initialize_CashInventory();
+		intialize_Item_Inventory();
+	}
+
+	public void displayMenu() {
+		System.out.println("Welcome");
+		System.out.println("MENU:");
+		System.out.println("1. Dairy Milk				Rs. 10/-");
+		System.out.println("2. Munch				Rs. 5/-");
+		System.out.println("3. Milky Bar				Rs. 15/-");
+		System.out.println("4. 5 Star				Rs. 5/-");
+		System.out.println(
+				"Please press the corresponding no. from the menu on the keypad and then quantity to get your chocolate ");
+
+	}
+
+	public boolean hasSufficientItem(int s_no, int qty) {
+		Item items_in[] = Item.values();
+		int avail_qty = items.get(items_in[s_no - 1]);

```

Comment:
you're not validating s_no, so it looks like you could easily get into an index out of bounds here

Reply:
same for all of the below methods